From ec33b15760b46d37fd50e6e8a3eab8de1559e9d6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=D0=BD=D0=B0=D0=B1?= <nabijaczleweli@nabijaczleweli.xyz>
Date: Sat, 26 Sep 2020 13:00:40 +0200
Subject: [PATCH] Fix <sidebar-{next,prev}-new>

Commit 7e743a9fe63bea20d20070e34c86837083790005 "Convert Sidebar entries
to ARRAY" incorrectly inverted a
"do {} while(!has_new && msg_unread == 0)" into
"if(has_new && msg_unread != 0)", whereas the correct De Morgan is
"if(has_new || msg_unread != 0)"

This fixes a regression against 20200821
---
 sidebar/functions.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sidebar/functions.c b/sidebar/functions.c
index 8a24281e90..7fea9c7983 100644
--- a/sidebar/functions.c
+++ b/sidebar/functions.c
@@ -72,7 +72,7 @@ static struct SbEntry **next_new(struct SidebarWindowData *wdata, size_t begin,
   struct SbEntry **sbep = NULL;
   ARRAY_FOREACH_FROM_TO(sbep, &wdata->entries, begin, end)
   {
-    if ((*sbep)->mailbox->has_new && (*sbep)->mailbox->msg_unread != 0)
+    if ((*sbep)->mailbox->has_new || (*sbep)->mailbox->msg_unread != 0)
       return sbep;
   }
   return NULL;
@@ -143,7 +143,7 @@ static struct SbEntry **prev_new(struct SidebarWindowData *wdata, size_t begin,
   struct SbEntry **sbep = NULL, **prev = NULL;
   ARRAY_FOREACH_FROM_TO(sbep, &wdata->entries, begin, end)
   {
-    if ((*sbep)->mailbox->has_new && (*sbep)->mailbox->msg_unread != 0)
+    if ((*sbep)->mailbox->has_new || (*sbep)->mailbox->msg_unread != 0)
       prev = sbep;
   }
 
